### ISSUE TITLE
Microsoft Graph DLLs errors

### ISSUE THREAD

**Issue by michalkubis** [rubit0/MsGraphForUnity#2]
Hi, I can't make Microsoft DLLs work in Unity.

![dll-errs](https://user-images.githubusercontent.com/47777142/105697679-2549c980-5f05-11eb-8b65-fdd50e1f24ff.png)

I tried to create blank project and add MS Graph DLLs from nuget one by one, but I have the same result.

Could you please help?

**UPDATE: I've created separated .net project and got all packages and dependencies via Nuget manager. Unity isn't complaining about DLLs anymore.**

**Comment by rubit0** [comment]
Have you enabled NetStandard? It should all work out of the box without you adding new dependencies. Still glad though you got it sorted out 👍

**Comment by michalkubis** [comment]
Here's what I see in Unity 2019.4.17f1 out of the box.
![default](https://user-images.githubusercontent.com/47777142/105749114-48936980-5f43-11eb-99b5-1306eef24ba7.png)

What's wrong is described here:
"System.Memory references strong named System.Runtime.CompilerServices.Unsafe Assembly references: 4.0.4.1 Found in project: 4.0.6.0."

I didn't bother going through all dependences, but I created blank Visual Studio console app, targeting .net461 and via nuget manager added Microsoft.Graph. Then build the project and used DLLs I found in BIN/Debug. Not sure if there's better way, but this did the job. I didn't need only one DLL, I think it was "System.ValueTuple.dll", as it's already in Unity.

**Comment by mattfedo-insight** [comment]
Not working in Version 2020.X - recompiled the DLLs and I also used Nuget for Unity to pull the DLLs - no dice. Same issue as above.

**Comment by rubit0** [comment]
Actually I was thinking of moving towards using the latest [NuGetForUnity](https://github.com/GlitchEnzo/NuGetForUnity/releases/tag/v3.0.2) since half of the problem this project solves is bringing in the correct packages. I hope to spent some time on that this weekend.

**Comment by mattfedo-insight** [comment]
Awesome @rubit0 - I did notice if you go to Project Settings > Other Settings and turn off Assembly Version Validation, all the errors go away and you can compile.  I believe that is only for Standalone PC/Mac though.  Switching to iOS / Android / UWP changes it all up again.  Much appreciated efforts here!

**Comment by rubit0** [comment]
Yeah I hope I can get it to work, if not I might have to fork MSGraph and bring in all the API Client stuff and then solve the sign-in in a lean way - but that would be a big lift.

**Comment by WarlordMSM** [comment]
@michalkubis can you explain your solution please? I am also stuck and need to fix it :(
